File: static/js/Ship.js
/**
  * class Ship
  * coding=UTF-8
  */

class Ship {
  constructor(lenght, horizontal) {
    this.m_lenght = lenght;
    this.m_lifes = lenght;
    this.m_horizontal = horizontal;
    this.m_hurt = [];
    this.x = 0;
    this.y = 0;
  }

  /**
   * @param x value of x
   */
  setX(x)
  {
    this.x=x;
  }

  /**
   * @param y
   */
  setY(y)
  {
    this.y = y;
  }


  /**
   * @return : true if ship is horizontal.
   */
  isHorizontal ()
  {
    return this.horizontal;
  }

  /**
   * @return : true is ship is dead.
   */
  isDead()
  {
    return this.m_.lifes < 0;
  }

  /**
   *     get's called from Battelship object
   *     @param int hitAt : the relative hit position of the ship
   *     @return  : False if ship is aready hitten at this position
   *     @author Christian
   *     @param hitAt
   */
  hit(hitAt) {
    //assert hitAt < this.m_.lenght,'unable to hit ship there max lenght '+str(this.m_.lenght)
    if (hitAt in this.hurt) {
      return false;
    } else {
      this.m_lifes--;
      return true;
    }
  }
}

File: backend/Ship.py
# coding=UTF-8

class Ship(object):

  """

  :version: 1.0
  :author: Christian
  """

  """ ATTRIBUTES



  lenght  (private)



  lives  (private)


  horizontal (private)

  """

  def __init__(self, lenght, horizontal):
      self.lenght = lenght
      self.lifes = self.lenght
      self.horizontal = horizontal
      self.hurt = []

  def setX(self, x):
      self.x = x

  def setY(self, y):
      self.y = y

  def hit(self, hitAt):
    assert hitAt < self.lenght, 'unable to hit ship there max lenght ' + str(self.lenght)
    """
    get's called from Battelship object

    @param int hitAt : the relative hit position of the ship
    @return  : False if ship is aready hitten at this position
    @author Christian
    """

    if hitAt in self.hurt:
        return False
    else:
        self.lifes -= 1
        return False

  def isHorizontal(self):
      """
      @return : true if ship is horizontal
      """
      return self.horizontal

  def isDead(self):
      return self.lifes < 0
